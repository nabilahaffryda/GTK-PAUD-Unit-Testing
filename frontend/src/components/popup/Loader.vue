<template>
  <v-dialog v-model="dialog" persistent width="300">
    <v-card color="primary" dark>
      <v-card-text class="pt-3">
        <span class="pt-2">Sedang dalam proses, silakan tunggu...</span>
        <v-progress-linear indeterminate color="white" class="mb-0"></v-progress-linear>
      </v-card-text>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: 'loader',
  props: {
    loading: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      dialog: this.loading,
    };
  },
  methods: {
    setDialog(value) {
      this.dialog = value;
    },
  },
  watch: {
    loading: 'setDialog',
  },
};
</script>

<style scoped></style>
