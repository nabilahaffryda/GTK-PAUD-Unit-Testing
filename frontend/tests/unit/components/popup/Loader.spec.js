import { mount, createLocalVue } from "@vue/test-utils";
import Loader from "@/components/popup/Loader.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('Loader.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })

    test('render loader', () => {
        const VLoader = {
            props: ['loading'],
            template: '<div><slot :loading="loading" /></div>'
        }
        const wrapper = mount(Loader, {
            localVue,
            vuetify,
            stubs: {
                VLoader
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
