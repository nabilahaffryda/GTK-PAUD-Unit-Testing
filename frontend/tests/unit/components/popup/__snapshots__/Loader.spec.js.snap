// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.vue render loader 1`] = `
<div role="dialog" class="v-dialog__container">
  <!---->
</div>
`;
